### PR TITLE
(CDAP-7704) Reported operational stats for HBase

### PR DESCRIPTION
**Pull request by bdmogal**
Jira: [CDAP-7704](https://issues.cask.co/browse/CDAP-7704)
Build: http://builds.cask.co/browse/CDAP-RUT327

### CODE REVIEW

File: cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseEntities.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  @Override
+  public int getSnapshots() {
+    return snapshots;
+  }
+
+  @Override
+  public void collect() throws IOException {
+    try (HBaseAdmin admin = new HBaseAdmin(conf)) {
+      namespaces = admin.listNamespaceDescriptors().length;
```
Comment (anwar6953) [inline cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseEntities.java:69]: I'm not sure how we should surface it to the users, but the number of hbase namespaces that this metric counts may be far less than the number of cdap namespaces (in the case that impersonation is configured for some namespaces, and the cdap system user doesn't have admin on the hbase instance).

Comment (bdmogal) [reply cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseEntities.java:69]: Good point. The purpose of these stats is not restricted to CDAP -- they were intended to count *all* HBase namespaces.

But I think there's a bigger point - what happens when some of these fail due to lack of permission for the CDAP user? These APIs were intended for cluster admins (not just CDAP, but admins for your entire cluster) to get a summary of the cluster. If the CDAP system user does not have the privileges to get some of this information, I think we may need an enhancement to optionally accept impersonation info for an admin and run these calls as that admin.

File: cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseInfo.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  @Override
+  public synchronized void collect() throws IOException {
+    try (HBaseAdmin admin = new HBaseAdmin(conf)) {
+      ClusterStatus clusterStatus = admin.getClusterStatus();
+      ServerName master = clusterStatus.getMaster();
+      String protocol = conf.getBoolean("hbase.ssl.enabled", false) &&
+        conf.get("hbase.http.policy").equals("HTTPS_ONLY") ?
+        "https" : "http";
+      webUrl = String.format("%s://%s:%s", protocol, master.getHostname(), conf.get("hbase.master.info.port"));
```
Comment (anwar6953) [inline cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseInfo.java:76]: what happens if an hbase install has authentication in front of its webapp? We log an error? 
We'll log it EVERY time that the bean is called to operate?

Comment (bdmogal) [reply cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseInfo.java:76]: we're not opening this connection, just constructing a URL for it, so this is just a string, and there shouldn't be any errors while doing this. All this does is chooses the right protocol, host and port. Once that is done, if the webapp is fronted by authentication, the user will have to log in after clicking on the link that points to this URL.

Comment (anwar6953) [reply cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseInfo.java:76]: I see. we only present the url to the user in the ui, and never attempt to connect to it. got it.

**Comment by anwar6953** [comment]
LGTM

File: cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseEntities.java (hunk 1 above)
Comment (bdmogal) [reply cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hbase/HBaseEntities.java:69]: https://issues.cask.co/browse/CDAP-7799

**Comment by bdmogal** [comment]
Build is green. Tested on a cluster. Merging.

**Pull request merged**